src/modules/shared/pages/innovation/actions/task-to-do-list.component.ts

```diff
@@ -0,0 +1,172 @@
+  }
+
+  ngOnInit(): void {
+
+    this.allTasksList.setFilters({
+      innovationId: this.innovationId,
+      fields: ['notifications'],
+      status: [
+        InnovationActionStatusEnum.OPEN,
+        InnovationActionStatusEnum.DONE,
+        InnovationActionStatusEnum.DECLINED,
+        InnovationActionStatusEnum.CANCELLED,
+      ],
+      allTasks: true,
+    });
+
+    
+    this.innovationsService.getActionsList(this.allTasksList.getAPIQueryParams()).subscribe((allTasksResponse) => {
+
+      this.processTaskList(allTasksResponse);
+
+      this.setPageStatus('READY');
+
+    });
+
+  }
+
+  processTaskList(taskList: InnovationActionsListDTO) {
+    for (let task of taskList.data) {
+      console.log(task)
+      if (this.shouldBeOnTopTable(task)) {
+        this.topList.data.push(task);
+      } else{
+        this.bottomList.data.push(task);
+      }
+    }
+    console.log("top list:")
+    console.log(this.topList);
+    console.log("bottom list:")
+    console.log(this.bottomList);
+    this.topList.count = this.topList.data.length;
+    this.bottomList.count = this.bottomList.data.length;
+  }
+
+  shouldBeOnTopTable(item: InnovationActionInfoDTO): boolean {
+    switch (this.getUserType()) {
+
+      case 'INNOVATOR':
+        return (
+          item.status == InnovationActionStatusEnum.OPEN
+        )
+
+      case 'ASSESSMENT':
+        return (!item.createdBy.organisationUnit)
+
+      case 'ACCESSOR':
+        
+        return item.createdBy.organisationUnit?.name ==
+        this.stores.authentication.getAccessorOrganisationUnitName()
+
+      return false;
+
+      case 'QUALIFYING_ACCESSOR':
+        return item.createdBy.organisationUnit?.name ==
+        this.stores.authentication.getAccessorOrganisationUnitName()
+
+      case 'ADMIN':
+        return true;
+
+      default:
+        return false;
+    }
+  }
+
+  getUserType() {
+    return this.stores.authentication.isAccessorType() ? UserRoleEnum.ACCESSOR : this.stores.authentication.getUserType();
+  }
+
+  tablesTitles(): { topTableTitle: string; bottomTableTitle: string } {
+    switch (this.getUserType()) {
+      case 'INNOVATOR':
+        return {
+          topTableTitle: `You have ${this.topList.count} tasks to do`,
+          bottomTableTitle: 'Previous tasks',
+        };
+      case 'ACCESSOR':
+        return {
+          topTableTitle: 'Tasks assigned by my organisation unit',
+          bottomTableTitle: 'Tasks assigned by others',
+        };
+      case 'QUALIFYING_ACCESSOR':
+        return {
+          topTableTitle: 'Tasks assigned by my organisation unit',
+          bottomTableTitle: 'Tasks assigned by others',
+        };
+      case 'ASSESSMENT':
+        return {
+          topTableTitle: 'Tasks assigned by Needs Assessment team',
+          bottomTableTitle: 'Tasks assigned by others',
+        };
+      case 'ADMIN':
+        return {
+          topTableTitle: '',
+          bottomTableTitle: '',
```

Comment:
It probably should have a title for the top one?

Reply:
Admin only shows one table, with no title, that's the reason for the empty strings. I left it like that in case something changes in the future, would there be a more correct way in this case?

Reply:
Okok, I thought it would have a title. You can leave it like this!